addons/website_slides/models/slide_channel.py

```diff
@@ -372,25 +371,40 @@ def _action_add_members(self, target_partners, **member_values):
 
         Returns the added 'slide.channel.partner's (! as sudo !)
         """
-        to_join = self._filter_add_members(target_partners, **member_values)
-        if to_join:
-            existing = self.env['slide.channel.partner'].sudo().search([
-                ('channel_id', 'in', self.ids),
-                ('partner_id', 'in', target_partners.ids)
-            ])
-            existing_map = dict((cid, list()) for cid in self.ids)
-            for item in existing:
-                existing_map[item.channel_id.id].append(item.partner_id.id)
-
-            to_create_values = [
-                dict(channel_id=channel.id, partner_id=partner.id, **member_values)
-                for channel in to_join
-                for partner in target_partners if partner.id not in existing_map[channel.id]
-            ]
-            slide_partners_sudo = self.env['slide.channel.partner'].sudo().create(to_create_values)
-            return slide_partners_sudo
+        target_partners = self._filter_target_members(target_partners, **member_values)
+        if target_partners:
+            to_join = self._filter_add_members(target_partners, **member_values)
+            if to_join:
+                existing = self.env['slide.channel.partner'].sudo().search([
+                    ('channel_id', 'in', self.ids),
+                    ('partner_id', 'in', target_partners.ids)
+                ])
+                existing_map = dict((cid, list()) for cid in self.ids)
+                for item in existing:
+                    existing_map[item.channel_id.id].append(item.partner_id.id)
+
+                to_create_values = [
+                    dict(channel_id=channel.id, partner_id=partner.id, **member_values)
+                    for channel in to_join
+                    for partner in target_partners if partner.id not in existing_map[channel.id]
+                ]
+                slide_partners_sudo = self.env['slide.channel.partner'].sudo().create(to_create_values)
+                return slide_partners_sudo
         return self.env['slide.channel.partner'].sudo()
 
+    def _filter_target_members(self, target_partners, **member_values):
+        filtered_target_partner_ids = []
+        public_users = self.env['res.users'].sudo()
+        public_groups = self.env.ref("base.group_public", raise_if_not_found=False)
+        if public_groups:
+            public_users = public_groups.sudo().with_context(active_test=False).mapped("users")
+        for target_partner in target_partners:
+            is_public = target_partner.sudo().with_context(active_test=False).user_ids in public_users if public_users else False
```

Comment:
Maybe you should check `_is_public()` on each user.

Reply:
I think I was in a parallel world when I did this branch..